src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -538,6 +544,60 @@ func (c controller) addRouterCAToClusterCA() bool {
 
 }
 
+// TODO: remove
+type OLMOperator struct {
+	name      string
+	namespace string
+	enabled   bool
+	status    string
+}
+
+// TODO: remove
+var olms = map[string]OLMOperator{
+	"lso": {
+		"local-storage-operator.4.6.0-202101300140.p0", "openshift-local-storage", true, "",
+	},
+}
+
+func (c controller) getOLMOperators() ([]OLMOperator, error) {
+	ret := []OLMOperator{}
+	operators, err := c.ic.GetClusterOLM(context.TODO())
+	if err != nil {
+		c.log.WithError(err).Warningf("Failed to connect to assisted service")
+		return ret, err
+	}
+	for _, operator := range *operators {
+		if *operator.Enabled && operator.Status != "Succeeded" {
+			ret = append(ret, olms[string(operator.OperatorType)])
+		}
+	}
+	return ret, nil
+}
+
+// TODO: remove
+func updateStatus(olm OLMOperator, phase string) {
+	olm.status = phase
+	olms[olm.name] = olm
```

Comment:
What about keeping track of conditions as well? It could be helpful to show users more information about potential failure.

Reply:
In current API design, there are `status` and `status_info` parameters. The `status` would reflect the `phase` and the `status_info` could reflect `message` from CSV's `status.message`. We could also store the conditions. But I think the only reason we store those informations is to decide whether the cluster should be "Succeeded" or "Degraded", not sure we need on the assisted service side to know all the conditions of the CSV. But definitelly it would be nice to log them.

Reply:
The status will be a translation of each operator (olm, cvo, co) status to a common status as the assisted-service will declare.
I think storing the condition will be too much information. The status_info will report the message for failed conditions or progress for cvo.